metaserver/src/main/scala/scala/meta/languageserver/Semanticdbs.scala

```diff
@@ -4,8 +4,25 @@ import java.nio.file.Files
 import com.typesafe.scalalogging.LazyLogging
 import org.langmeta.internal.semanticdb.schema.Database
 import org.langmeta.io.AbsolutePath
+import scala.meta.interactive.InteractiveSemanticdb
+import scala.meta.parsers.ParseException
+import scala.tools.nsc.interactive.Global
+import langserver.types.VersionedTextDocumentIdentifier
+import ScalametaEnrichments._
 
 object Semanticdbs extends LazyLogging {
+  def loadFromTextDocument(
+    compiler: Global,
+    td: VersionedTextDocumentIdentifier,
+    content: String
+  ): scala.meta.semanticdb.Database = {
+    val documents = try {
+      List(InteractiveSemanticdb.toDocument(compiler, content, Uri.toPath(td.uri).get.toLanguageServerUri, 10000))
```

Comment:
this filename "dance" seems necessary because from my early tests `file:///User/...` crashes the PC :O 
This may need further investigation.

Reply:
I'm still not really happy with how we organize URI/AbsolutePath, it seems sometimes we want URI/AbsolutePath/String/... and we do conversions back and forth all over the place.